src/main/java/org/zalando/intellij/swagger/service/SwaggerFileService.java

```diff
@@ -27,27 +28,34 @@
     final private ExecutorService executorService = Executors.newSingleThreadExecutor();
 
     public Optional<Path> convertSwaggerToHtml(@NotNull final PsiFile file) {
-        if (fileDetector.isMainSwaggerJsonFile(file) || fileDetector.isMainOpenApiJsonFile(file)) {
-            return convertSwaggerToHtml(file, file.getText());
-        } else {
-            return convertSwaggerToHtml(file, yamlToJson(file.getText()));
+        try {
+            return convertSwaggerToHtml(file, convertToJsonIfNecessary(file));
+        } catch (Exception e) {
+            notifyFailure(e);
+            return Optional.empty();
         }
     }
 
     public void convertSwaggerToHtmlAsync(@NotNull final PsiFile file) {
-        if (fileDetector.isMainSwaggerJsonFile(file) || fileDetector.isMainOpenApiJsonFile(file)) {
-            executorService.submit(() -> {
-                convertSwaggerToHtml(file, file.getText());
-            });
-        } else {
-            executorService.submit(() -> {
-                final String contentAsJson = yamlToJson(file.getText());
-                convertSwaggerToHtml(file, contentAsJson);
-            });
-        }
+        executorService.submit(() -> {
+            try {
+                convertSwaggerToHtml(file, convertToJsonIfNecessary(file));
+            } catch (Exception e) {
+                notifyFailure(e);
+            }
+        });
     }
 
-    private Optional<Path> convertSwaggerToHtml(final PsiFile file, final String contentAsJson) {
+    private void notifyFailure(final Exception exception) {
+        Notification notification = new Notification(
+                "Swagger UI",
+                "Could not generate Swagger UI",
+                exception.getMessage(), NotificationType.WARNING);
```

Comment:
Purely æsthetic:
If you "bullet list" your argument (1 per line), you should probably do it all the way down (put the last one also on its own line).

Reply:
Good point, I created an issue https://github.com/zalando/intellij-swagger/issues/196 to add formatting to the whole codebase.

Reply:
@semonte : In my _personal_ coding style, the only exception to this I apply is when there is formatting involved. But that would not really happen in Kotlin since there is interpolation. But in something like _GO_, I would write the format on a line and the parameters on one line.

```go
fmt.Printf(
    "error parsing %q into a value of type %t in the context of %s",
    rawValue, parsedValue, currentContext)
```